libdebug/libdebug.py

```diff
@@ -1122,8 +1213,18 @@ def __threaded_wait(self):
         else:
             liblog.debugger("Waiting for process %d to stop.", self.context.process_id)
 
-        while self.interface.wait():
-            self.interface.cont()
+        while True:
+            self.context._resume = None
+            self.interface.wait()
+            value = self.context._resume
+            match value:
```

Comment:
The three states are necessary for managing concurrency among threads. It's crucial to distinguish when the variable has been explicitly set to False (a strong decision) from when it hasn't been set because the action to take is unclear (the default decision).

The variable is not returned because self.interface.wait() includes numerous inner functions that need to modify this value. Passing it as a parameter and then returning it made the code unreadable and difficult to maintain. This approach was problematic in the old implementation and is worse in the current one, due to the three states and added logic. I attempted to keep the return value, but this resulted in a solution that was confusing and anti-design patterns in a way.

The while + match is a placeholder, I do not like the solution.

Reply:
2753b22214e15b093abcab1380b6128f23c6206c